src/object.cpp

```diff
@@ -1104,6 +1104,140 @@ void Object::Process(Functor *functor, FunctorParams *functorParams, Functor *en
     }
 }
 
+void Object::Process(MutableFunctor &functor, int deepness, bool skipFirst)
+{
+    if (functor.GetCode() == FUNCTOR_STOP) {
+        return;
+    }
+
+    // Update the current score stored in the document
+    this->UpdateDocumentScore(functor.GetDirection());
+
+    if (!skipFirst) {
+        FunctorCode code = this->Visit(functor);
+        functor.SetCode(code);
+    }
+
+    // do not go any deeper in this case
+    if (functor.GetCode() == FUNCTOR_SIBLINGS) {
+        functor.SetCode(FUNCTOR_CONTINUE);
+        return;
+    }
+    else if (this->IsEditorialElement()) {
+        // since editorial object doesn't count, we increase the deepness limit
+        ++deepness;
+    }
+    if (deepness == 0) {
+        // any need to change the functor m_returnCode?
+        return;
+    }
+    --deepness;
+
+    if (!this->SkipChildren(functor.VisibleOnly())) {
+        // We need a pointer to the array for the option to work on a reversed copy
+        ArrayOfObjects *children = &m_children;
+        Filters *filters = functor.GetFilters();
+        if (functor.GetDirection() == BACKWARD) {
+            for (ArrayOfObjects::reverse_iterator iter = children->rbegin(); iter != children->rend(); ++iter) {
+                // we will end here if there is no filter at all or for the current child type
+                if (this->FiltersApply(filters, *iter)) {
+                    (*iter)->Process(functor, deepness);
+                }
+            }
+        }
+        else {
+            for (ArrayOfObjects::iterator iter = children->begin(); iter != children->end(); ++iter) {
+                // we will end here if there is no filter at all or for the current child type
+                if (this->FiltersApply(filters, *iter)) {
+                    (*iter)->Process(functor, deepness);
+                }
+            }
+        }
+    }
+
+    if (functor.ImplementsEndInterface() && !skipFirst) {
+        FunctorCode code = this->VisitEnd(functor);
+        functor.SetCode(code);
```

Comment:
Why not setting the `code` directly in `Visit` or `VisitEnd` methods? Then you would do just
```cpp
this->VisitEnd(functor);
```

Reply:
We would have to set the code in each override of `Visit(End)`, now `Accept(End)`. So we would trade setting it twice in `Process` against setting it 250 times in `Accept`.